beacon-chain/verification/data_column.go

```diff
@@ -282,6 +285,61 @@ func (dv *RODataColumnsVerifier) ValidProposerSignature(ctx context.Context) (er
 	return nil
 }
 
+// getVerifyingState returns a state that is compatible with the column sidecar and can be used to verify signature and proposer index.
+// The returned state is guaranteed to be at the same epoch as the data column's epoch, and have the same randao mix and active
+// validator indices as the data column's parent state advanced to the data column's slot.
+func (dv *RODataColumnsVerifier) getVerifyingState(ctx context.Context, dataColumn blocks.RODataColumn) (state.ReadOnlyBeaconState, error) {
+	headRoot, err := dv.hsp.HeadRoot(ctx)
+	if err != nil {
+		return nil, err
+	}
+	parentRoot := dataColumn.ParentRoot()
+	dataColumnSlot := dataColumn.Slot()
+	dataColumnEpoch := slots.ToEpoch(dataColumnSlot)
+	headSlot := dv.hsp.HeadSlot()
+	headEpoch := slots.ToEpoch(headSlot)
+
+	// Use head if it's the parent
+	if bytes.Equal(parentRoot[:], headRoot) {
```

Comment:
Since these are already fixed size byte arrays, you can directly compare them (eg `if parentRoot == headRoot`) which I think is more streamlined to read and should be about the same perf if not marginally faster, at least according to a quick benchmark.

Reply:
headRoot is a slice